contracts/V1/StateChannelDiamondProxy/DisputeFraudProofFacet.sol

```diff
@@ -151,18 +154,24 @@ contract DisputeFraudProofFacet is StateChannelCommon {
         DisputeNotLatestStateProof memory disputeNotLatestStateProof =
             abi.decode(encodedFraudProof, (DisputeNotLatestStateProof));
         Block memory newerBlock = abi.decode(disputeNotLatestStateProof.encodedBlock, (Block));
-        Block memory latestBlock = _getLatestBlock(dispute.stateProof);
+        (bool hasBlock, Block memory latestBlock) = _getLatestBlock(dispute.stateProof);
 
-        // Check channelId
-        if (!_areDisputeAndBlockSameChannel(dispute, newerBlock) || !_areBlocksSameChannel(newerBlock, latestBlock)) {
-            revert();
-        }
+        // Check newBlock same channelId
+        if (!_areDisputeAndBlockSameChannel(dispute, newerBlock)) revert();
+        // Check newBlock same forkId
+        if (!_areDisputeAndBlockSameFork(dispute, newerBlock)) revert();
 
-        // Check forkId
-        if (!_areDisputeAndBlockSameFork(dispute, newerBlock) || !_areBlocksSameFork(newerBlock, latestBlock)) revert();
+        if (hasBlock) {
+            // Check latestBlock and newerBlock same channelId
+            if (!_areBlocksSameChannel(newerBlock, latestBlock)) revert();
 
-        // Check is block newer
-        if (_getBlockHeight(newerBlock) <= _getBlockHeight(latestBlock)) revert();
+            // Check latestBlock and newerBlock same forkId
+            if (!_areBlocksSameFork(newerBlock, latestBlock)) revert();
+
+            // Check is block newer
+            if (_getBlockHeight(newerBlock) <= _getBlockHeight(latestBlock)) revert();
+        }
+        // if !hasBlock -> latestState should be genesis state -> if the disputer signed any block this proof is valid
```

Comment:
Changes in this file are mostly `_getLatestBlock(stateProof)` now accounting for stateProof being empty (can create a dispute with the `latestState == genesisState`)